File: Milktea/LeetCode/Code/JavaScript/2654.js
var minOperations = function (nums) {
  let n = nums.length;
  let countOne = 0;
  for (let i = 0; i < n; i++) {
    countOne += nums[i] === 1 ? 1 : 0;
  }
  if (countOne > 0) {
    return n - countOne;
  }
  let res = Number.MAX_SAFE_INTEGER;
  for (let i = 0; i < n; i++) {
    let num = nums[i];
    for (let j = i + 1; j < n; j++) {
      num = gcd(num, nums[j]);
      if (num == 1) {
        res = Math.min(res, j - i);
        break;
      }
    }
  }
  if (res == Number.MAX_SAFE_INTEGER) {
    return -1;
  }
  return res + n - 1;
};

var gcd = function (a, b) {
  if (b == 0) {
    return a;
  }
  return gcd(b, a % b);
};

File: Milktea/LeetCode/Code/Python/2654.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        countOne = nums.count(1)

        if countOne: return n - countOne
        res = inf

        for i in range(n):
            num = nums[i]
            for j in range(i + 1, n):
                num = gcd(num, nums[j])
                if num == 1:
                    res = min(res, j - i)

        if res == inf: return -1
        return res + n - 1